components/Workout/index.js

```diff
@@ -0,0 +1,101 @@
+import styled from "styled-components";
+import ModalDelete from "../ModalDelete";
+import Link from "next/link";
+import WorkoutPreview from "../WorkoutPreview";
+import { useState } from "react";
+import { toast } from "react-toastify";
+
+export default function Workout({ workouts, onDeleteWorkout, exercises }) {
+  const [showModal, setShowModal] = useState(false);
+  const [workoutIdToDelete, setWorkoutIdToDelete] = useState(0);
+
+  const handleDelete = (id) => {
+    setWorkoutIdToDelete(id);
+    setShowModal(true);
+  };
+
+  function handleDeleteCancel() {
+    setShowModal(false);
+    setWorkoutIdToDelete(null);
+  }
+
+  function successMessage() {
+    toast.success("Workout deleted successfully!");
+  }
+
+  function handleDeleteConfirm() {
+    onDeleteWorkout(workoutIdToDelete);
+    setShowModal(false);
+    setWorkoutIdToDelete(null);
+    successMessage();
+  }
+  return (
+    <>
+      {workouts.map((workout) => (
+        <WorkoutItem key={workout.id}>
+          <Actions>
+            <DeleteButton
+              type="button"
+              onClick={() => handleDelete(workout.id)}
+            >
+              ✘ Delete
+            </DeleteButton>
+            <EditLink href={`workouts/${workout.id}/edit`}>✎ Edit</EditLink>
+          </Actions>
+          <WorkoutPreview
+            name={workout.name}
+            workoutExercises={workout.exercises}
+            exercises={exercises}
+            workouts={workouts}
+          />
+          {showModal && (
+            <ModalDelete
+              id={workoutIdToDelete}
+              onConfirm={handleDeleteConfirm}
+              onCancel={handleDeleteCancel}
+            />
+          )}
+        </WorkoutItem>
+      ))}
+    </>
```

Comment:
```suggestion
          />
        </WorkoutItem>
      ))}
      {showModal && (
        <ModalDelete
          id={workoutIdToDelete}
          onConfirm={handleDeleteConfirm}
          onCancel={handleDeleteCancel}
        />
      )}
    </>
```
Right now the `ModalDelete` component is rendered inside the `.map` loop for each `WorkoutItem`.  I'd suggest moving `ModalDelete` outside the `.map` loop to avoid rendering multiple instances of it.

Also, `handleDeleteConfirm` doesn't expect a parameter, so you don't necessarily have to pass down `workoutIdToDelete`